runtime/sema/type_tags.go

```diff
@@ -670,7 +674,8 @@ func findSuperTypeFromUpperMask(joinedTypeTag TypeTag, types []Type) Type {
 		restrictedTypeMask,
 		transactionTypeMask,
 		interfaceTypeMask,
-		functionTypeMask:
+		functionTypeMask,
+		inclusiveRangeTypeMask:
```

Comment:
Do we want InclusiveRange to be covariant?

Reply:
What is covariant and what are the implications of it being one?

Reply:
So in this case, a covariant would mean, the `InclusiveRange<Int8>` value can be assigned to `InclusiveRange<Integer>`.

It would also mean that if there is an array with two (or more) values with different range types, say `var x = [InclusiveRange<Int8>(), InclusiveRange<Int16>()]`, the inferred type for `x` will be the covariant-super-type `InclusiveRange<Integer>`.

I feel we don't want to support this. But WDYT?
Also might be good to add some tests, to verify the behavior, if there aren't any already

Reply:
> I feel we don't want to support this. WDYT?

I am actually not sure of the pros and cons of supporting this. So it is fine for me if we don't support it.

cc: @turbolent if you have any thoughts.

Reply:
> Also might be good to add some tests, to verify the behavior, if there aren't any already

I've added test cases in `TestRuntimeScriptParameterTypeValidation`.